applications/product/groovyScripts/facility/facility/CountFacilityInventoryByProduct.groovy

```diff
@@ -55,165 +57,168 @@ rows = [] as ArrayList
-        salesUsageViewEntity.addAlias("OI", "productId")
-        salesUsageViewEntity.addAlias("OH", "statusId")
-        salesUsageViewEntity.addAlias("OH", "orderTypeId")
-        salesUsageViewEntity.addAlias("OH", "orderDate")
-        salesUsageViewEntity.addAlias("ItIss", "inventoryItemId")
-        salesUsageViewEntity.addAlias("ItIss", "quantity")
-        salesUsageViewEntity.addAlias("InvIt", "facilityId")
+        salesUsageViewEntity = new DynamicViewEntity().with {
+            addMemberEntity('OI', 'OrderItem')
+            addMemberEntity('OH', 'OrderHeader')
+            addMemberEntity('ItIss', 'ItemIssuance')
+            addMemberEntity('InvIt', 'InventoryItem')
+            addViewLink('OI', 'OH', false, ModelKeyMap.makeKeyMapList('orderId'))
+            addViewLink('OI', 'ItIss', false, ModelKeyMap.makeKeyMapList('orderId', 'orderId', 'orderItemSeqId', 'orderItemSeqId'))
+            addViewLink('ItIss', 'InvIt', false, ModelKeyMap.makeKeyMapList('inventoryItemId'))
+            addAlias('OI', 'productId')
+            addAlias('OH', 'statusId')
+            addAlias('OH', 'orderTypeId')
+            addAlias('OH', 'orderDate')
+            addAlias('ItIss', 'inventoryItemId')
+            addAlias('ItIss', 'quantity')
+            addAlias('InvIt', 'facilityId')
+        }
 
         // Construct a dynamic view entity to search against for production usage quantities
-        productionUsageViewEntity = new DynamicViewEntity()
-        productionUsageViewEntity.addMemberEntity("WEIA", "WorkEffortInventoryAssign")
-        productionUsageViewEntity.addMemberEntity("WE", "WorkEffort")
-        productionUsageViewEntity.addMemberEntity("II", "InventoryItem")
-        productionUsageViewEntity.addViewLink("WEIA", "WE", false, ModelKeyMap.makeKeyMapList("workEffortId"))
-        productionUsageViewEntity.addViewLink("WEIA", "II", false, ModelKeyMap.makeKeyMapList("inventoryItemId"))
-        productionUsageViewEntity.addAlias("WEIA", "quantity")
-        productionUsageViewEntity.addAlias("WE", "actualCompletionDate")
-        productionUsageViewEntity.addAlias("WE", "workEffortTypeId")
-        productionUsageViewEntity.addAlias("II", "facilityId")
-        productionUsageViewEntity.addAlias("II", "productId")
+        productionUsageViewEntity = new DynamicViewEntity().with {
+            addMemberEntity('WEIA', 'WorkEffortInventoryAssign')
+            addMemberEntity('WE', 'WorkEffort')
+            addMemberEntity('II', 'InventoryItem')
+            addViewLink('WEIA', 'WE', false, ModelKeyMap.makeKeyMapList('workEffortId'))
+            addViewLink('WEIA', 'II', false, ModelKeyMap.makeKeyMapList('inventoryItemId'))
+            addAlias('WEIA', 'quantity')
+            addAlias('WE', 'actualCompletionDate')
+            addAlias('WE', 'workEffortTypeId')
+            addAlias('II', 'facilityId')
+            addAlias('II', 'productId')
+        }
```

Comment:
Need to return `it` for assignment to `productionUsageViewEntity`

Reply:
Done Thanks !